File: android-app/app/src/main/java/com/grocery/app/utils/DeliveryUtils.java
package com.grocery.app.utils;

import android.location.Location;

public class DeliveryUtils {
    // Shop location coordinates (Replace with actual shop coordinates)
    private static final double SHOP_LAT = 28.7041;
    private static final double SHOP_LNG = 77.1025;

    public static double calculateDistance(double customerLat, double customerLng) {
        float[] results = new float[1];
        Location.distanceBetween(SHOP_LAT, SHOP_LNG, customerLat, customerLng, results);
        return results[0] / 1000.0; // Convert meters to kilometers
    }

    public static double getDeliveryCharge(double distanceKm) {
        if (distanceKm <= 5.0) return 20.0;
        if (distanceKm <= 10.0) return 40.0;
        if (distanceKm <= 20.0) return 80.0;
        if (distanceKm <= 50.0) return 150.0;
        return -1.0; // Not serviceable
    }
}

File: android-app/app/src/main/java/com/grocery/app/utils/DeliveryUtils.kt
package com.grocery.app.utils

import android.location.Location

object DeliveryUtils {
    // Shop location coordinates (Replace with actual shop coordinates)
    private const val SHOP_LAT = 28.7041
    private const val SHOP_LNG = 77.1025

    fun calculateDistance(customerLat: Double, customerLng: Double): Double {
        val results = FloatArray(1)
        Location.distanceBetween(SHOP_LAT, SHOP_LNG, customerLat, customerLng, results)
        return results[0] / 1000.0 // Convert meters to kilometers
    }

    fun getDeliveryCharge(distanceKm: Double): Double {
        return when {
            distanceKm <= 5.0 -> 20.0
            distanceKm <= 10.0 -> 40.0
            distanceKm <= 20.0 -> 80.0
            distanceKm <= 50.0 -> 150.0
            else -> -1.0 // Not serviceable
        }
    }
}
